stl/inc/xlocnum

```diff
@@ -1292,18 +1292,23 @@ protected:
         _OutIt _Dest, ios_base& _Iosbase, _Elem _Fill, double _Val) const { // put formatted double to _Dest
         string _Buf;
         char _Fmt[8];
-        bool _Isfixed         = (_Iosbase.flags() & ios_base::floatfield) == ios_base::fixed;
-        streamsize _Precision = _Iosbase.precision() <= 0 && !_Isfixed ? 6 : _Iosbase.precision(); // desired precision
-        size_t _Bufsize       = static_cast<size_t>(_Precision);
+        const bool _Isfixed = (_Iosbase.flags() & ios_base::floatfield) == ios_base::fixed;
+        const bool _Ishex   = (_Iosbase.flags() & ios_base::floatfield) == (ios_base::fixed | ios_base::scientific);
+        const streamsize _Precision =
+            _Ishex ? (DBL_MANT_DIG + 2) / 4
+                   : _Iosbase.precision() <= 0 && !_Isfixed ? 6 : _Iosbase.precision(); // desired precision
+        size_t _Bufsize = static_cast<size_t>(_Precision);
         if (_Isfixed && 1e10 < _CSTD fabs(_Val)) { // f or F format
             int _Ptwo;
             (void) _CSTD frexp(_Val, &_Ptwo);
             _Bufsize += _CSTD abs(_Ptwo) * 30103L / 100000L;
         }
 
         _Buf.resize(_Bufsize + 50); // add fudge factor
-        const auto _Ngen = static_cast<size_t>(_CSTD sprintf_s(
-            &_Buf[0], _Buf.size(), _Ffmt(_Fmt, 0, _Iosbase.flags()), static_cast<int>(_Precision), _Val));
+        const auto _Ngen = static_cast<size_t>(
+            _Ishex ? _CSTD sprintf_s(&_Buf[0], _Buf.size(), _Ffmt(_Fmt, 0, _Iosbase.flags(), false), _Val)
+                   : _CSTD sprintf_s(&_Buf[0], _Buf.size(), _Ffmt(_Fmt, 0, _Iosbase.flags(), true),
+                       static_cast<int>(_Precision), _Val));

```

Comment:
another way to reduce (source) code size is to just set some temporary in this if statement and only write sprintf_s once.

Also I generally don't love ternaries, although this one isn't _that_ bad